atlp-rwanda/Dominators-Phantom-BE
# Bg fixing routes coordinates

### What does this PR do?
- Adds coordinates of origin and destination in the model
- Fixes bugs in CRUD of routes
- Creates seeder for routes
- Fixes the seeder of bus to use the routeId which exists in DB
### Description of Task to be completed?

### How should this be manually tested?
- After cloning the repo run `npm i` to install all packages
- For Windows users refer an extra step for installing [`redis`](https://www.youtube.com/watch?v=188Fy-oCw4w&ab_channel=ProgrammingKnowledge2)
- use postman to test different endpoints present in [`index.js`](https://github.com/atlp-rwanda/Dominators-Phantom-BE/blob/develop/src/routes/index.js)
- Run `npm run migrate` to migrate all models in the local database
- Run `npm run seed` to populate the tables with seeders' values
- Run `npm run dev` to start the server manually
- use `http://localhost:3000/api/v1/login` to log in by using admin credentials present in the seeder of the user table. (Change the port **3000** to adhere to your chosen port)
- After login and adding Authorisation in headers, insert the URL with the port and the method e.g., `http://localhost:3000/api/v1/routes` and `POST`
- The request body includes  two coordinates which consist of latitudes and longitudes of both origin and destination
``` 
  {
    "origin": "Huye",
    "destination": "Kigali",
    "distance": "200km",
    "code": "305",
    "fromLatitude": 10.234567,
    "fromLongitude": 32.05556,
    "toLatitude": 1.234567,
    "toLongitude": 32.03456
}
```
- The expected response should look like this
```
{
    "Message": "success",
    "code": 200,
    "record": "Record created successfully"
}
```

### Any background context you want to provide?
The old seeder of the bus was deleted from this branch to avoid conflicts while migrating and seeding the tables. It is recommended that the one in `develop` called `20220603114223-BusSeed.js` can be deleted too. 
### What are the relevant pivotal tracker stories?

### Screenshots (if appropriate)

### Questions:
